RemoteInspectorMessageParser: Checking if vector is empty is redundant

LumioseSil opened the pull request:
#### 1fb67415dc152397ed426ecc1acc5c7e6fcb663e
<pre>
RemoteInspectorMessageParser: Checking if vector is empty is redundant
<a href="https://bugs.webkit.org/show_bug.cgi?id=255532">https://bugs.webkit.org/show_bug.cgi?id=255532</a>

Reviewed by Fujii Hironori.

Because we only operate on it if the size of the array is more than a
uint32_t, checking if it is empty is redundant.

* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorMessageParser.cpp:
  (Inspector::MessageParser::parse):

Canonical link: <a href="https://commits.webkit.org/263141@main">https://commits.webkit.org/263141@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e641cf6155debc433de34637d86c582c5b6be0f3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3696 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3781 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3887 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5123 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3979 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3858 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3786 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/3203 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3740 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3967 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3338 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4957 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1473 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3307 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/3363 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/3033 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3281 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3366 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/4725 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/3472 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3749 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3034 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/3769 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3282 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3307 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/952 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/3316 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/3861 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/424 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3559 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1065 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

fujii reviewed (approved):
Thank you. LGTM.

The title sounds too generic. You should add a prefix, for examepl "RemoteInspectorMessageParser:" or "[socket] " or something else.

> From: Rose <83477269+AtariDreams at users.noreply.github.com>

It'd better to set your name and mail address.
https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration

LumioseSil committed:
RemoteInspectorMessageParser: Checking if vector is empty is redundant
https://bugs.webkit.org/show_bug.cgi?id=255532

Reviewed by Fujii Hironori.

Because we only operate on it if the size of the array is more than a
uint32_t, checking if it is empty is redundant.

* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorMessageParser.cpp:
  (Inspector::MessageParser::parse):

Canonical link: https://commits.webkit.org/263141@main
